pset-functions.js

```diff
@@ -91,6 +155,28 @@ console.log('Problem 2:')
 
 // Add your code below this line
 
+
+var input = "olly olly in come free";
+var arrayOfWords = input.split(/\s+/);
+
+var wordCounts = Object.create(null);
+
+for (i = 0; i < arrayOfWords.length; i++) {
+    var word = arrayOfWords[i];
+    // add count of one if new word
+    if (!wordCounts[word]) {
+        wordCounts[word] = 1;
+    } else {
+    //add if existing
+        wordCounts[word]++;
+    }
+}
+
+var result = Object.keys(wordCounts).map(function(word) {
+  console.log(`${word} : ${wordCounts[word]}`)
+});
+
+
 // Add your code above this line
```

Comment:
@thetripler Good job here and this problem. Your looping through each word and incrementing by 1 if the word exists and setting the value to 1 if the word doesn't exist. I have a suggestion to avoid using the `var` keyword for creating variables and instead you can  use `let` or `const` for creating variables. Also for the wordCounts variable instead of using `Object.create` you can just use an empty object `{}`.

Take a look at this solution [here](https://github.com/jsr20190826/hw-02-functions-solution/blob/a4d789161886ff115657133c279062df8a4ca72d/pset-functions.js#L103)